src/mpl/src/shm/mpl_shm_mmap.c

```diff
@@ -85,6 +85,10 @@ static inline int MPL_shm_seg_create_attach_templ(MPL_shm_hnd_t hnd, intptr_t se
         do {
             rc = (int) write(lhnd, "", 1);
         } while ((rc == -1) && (errno == EINTR));
+        if (rc == -1) {
+            perror("shm segment creation failed");
```

Comment:
That's fair. I can remove the print. It would be nice to convert `errno` to string somehow and return it to the caller so they can choose to print it, but need to think how to do that cleanly across the interface.